src/main.cpp

```diff
@@ -13,36 +13,42 @@ int main(int argc, char **argv) {
   UserArgumentStruct UserArguments;
   std::string Broker;
 
-  App.add_option("-g, --go", UserArguments.GoBack,
-                 "How many records back to show. Otherwise "
-                 "retrieve entire topic. Mutually "
-                 "exclusive with \"--Offset\"");
+  App.add_option(
+      "-g, --go", UserArguments.GoBack,
+      "How many records back to show from partition \"-p\". Otherwise "
+      "retrieve entire topic. Mutually "
+      "exclusive with \"--Offset\"");
   App.add_option("-t, --topic", UserArguments.Name,
                  "Show records of specified topic");
   App.add_option("-b,--broker", Broker, "Hostname or IP of Kafka broker");
   App.add_option("-o,--offset", UserArguments.OffsetToStart,
                  "Start consuming from an offset. Otherwise retrieve entire "
                  "topic. Mutually exclusive with \"--go\"");
+  App.add_option("-p,--partition", UserArguments.PartitionToConsume,
+                 "Partition to retrieve messages from");
 
   App.add_flag("-a, --all", UserArguments.ShowAllTopics,
                "Show a list of topics");
   App.add_flag("-C, --consumer", UserArguments.ConsumerMode,
                "Run the program in the consumer mode");
   App.add_flag("-l, --list", UserArguments.MetadataMode,
                "Run the program in the metadata mode");
-  App.add_flag("-p, --partitions", UserArguments.ShowPartitionsOffsets,
-               "Show offsets for partitions of given topic \"-t\"");
+  //  App.add_flag("-p, --partitions", UserArguments.ShowPartitionsOffsets,
```

Comment:
Try not to leave dead code in. This information could be displayed as part of the metadata option in the future, could you create a ticket for that?